tmdsimpy/nlforces/unilateral_spring.py

```diff
@@ -55,28 +56,30 @@ def force(self, X):
             DESCRIPTION.
 
         """
-        raise Exception('Test this function before using it')
-        
-        unl = self.Q @ X
+        # raise Exception('Test this function before using it')
+        unl = self.Q @ X 
         
         fnl = np.maximum(self.k*(unl - self.delta) - self.Npreload, -self.Npreload)
         
         F = self.T @ fnl
-        
+
         mask = np.greater(fnl, -self.Npreload)
-        
-        dFdX = self.T @ (mask*self.k) @ self.Q
+        temp = (mask*self.k)
```

Comment:
This temp can be combined into the next line as the previous version of the code did.